src/app/content/utils.ts

```diff
@@ -57,13 +58,14 @@ export function getContentPageReferences(book: ArchiveBook, page: ArchivePage) {
   return matches;
 }
 
-export const parseContents = (book: Book, contents: Array<ArchiveTree | ArchiveTreeNode>) => {
-  contents.map((subtree) => {
```

Comment:
This `map` used to return the (modified) subtree, but the result was not used. So it must have worked by modifying the arguments in place during execution.